subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -360,6 +371,66 @@ static struct bass_recv_state_internal *get_free_recv_state(void)
 		}
 	}
 
+	LOG_DBG("No inactive receive state available");
+
+#if defined(CONFIG_BT_BAP_SCAN_DELEGATOR_OVERWRITE_OLDEST_RECV_STATE)
+	/* Free the oldest updated receive state that is not currently being synced
+	 */
+
+	struct bass_recv_state_internal *oldest_inactive_state = NULL;
+
+	for (size_t i = 0U; i < ARRAY_SIZE(scan_delegator.recv_states); i++) {
+		struct bass_recv_state_internal *current_state =
+					&scan_delegator.recv_states[i];
```

Comment:
I would suggest to put the code here that finds the oldest inactive state into another function for readability.